src/reducers/filterReducer.js

```diff
@@ -1,15 +1,83 @@
 import * as filterActions from 'actions/filterActions'
+import { combineReducers } from 'redux'
 
-export const filter = (defaultFilter) => (state = defaultFilter, action) => {
+export const incidentId = (defaultIncidentId) => (state = defaultIncidentId, action) => {
   switch (action.type) {
     case filterActions.CLEAR_EVENT_FILTER_INCIDENTID:
-      const {incidentId, ...newFilter} = action.filter
-      return newFilter
+      return null
     case filterActions.CHANGE_EVENT_FILTER:
-      return {...state, eventTypes: [], ticketId: null, ...action.filter}
+      if (action.filter.incidentId) {
+        return action.filter.incidentId
+      } else return state
     default:
       return state
   }
 }
 
+export const eventTypes = (defaultEventTypes) => (state = defaultEventTypes, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.eventTypes) {
+        return action.filter.eventTypes
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const startTime = (defaultStartTime) => (state = defaultStartTime, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.startTime) {
+        return action.filter.startTime
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const endTime = (defaultEndTime) => (state = defaultEndTime, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.endTime) {
+        return action.filter.endTime
+      } else return state
+    default:
+      return state
+  }
+}
+
+export const dataSearch = (defaultDataSearch) => (state = defaultDataSearch, action) => {
+  switch (action.type) {
+    case filterActions.CHANGE_EVENT_FILTER:
+      if (action.filter.DataSearch) {
```

Comment:
We want to be able to clear DataSearch to remove the filter and return to the (relatively) unfiltered view. This check prevents that. I'm not sure whether this check is necessary in any of the reducers. Checking for strict equality against state may be a better way to ensure we're changing the state when we want to and not changing it otherwise.

```JavaScript
case filterActions.CHANGE_EVENT_FILTER: 
  return action.filter.DataSearch === state
    ? state
    : action.filter.DataSearch
```